### PR TITLE
refactor: use absl::StrJoin in simple cases

### PR DESCRIPTION
**Pull request by remyabel**
Part of the work for: #5121

This uses `absl::StrJoin` where it does not require too much refactoring.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5159)

### CODE REVIEW

File: google/cloud/bigtable/examples/bigtable_examples_common.cc [hunk 1]
```diff
@@ -171,10 +172,7 @@ google::cloud::bigtable::examples::Commands::value_type MakeCommandEntry(
         argv.size() != 3 + args.size()) {
       std::ostringstream os;
       os << name << " <project-id> <instance-id> <table-id>";
-      char const* sep = " ";
-      for (auto const& a : args) {
-        os << sep << a;
-      }
+      os << absl::StrJoin(args, " ");
```
Comment (devbww) [inline google/cloud/bigtable/examples/bigtable_examples_common.cc:175]: These look like they drop the initial separator.

File: google/cloud/storage/notification_metadata.cc [hunk 2]
```diff
@@ -55,21 +56,15 @@ std::ostream& operator<<(std::ostream& os, NotificationMetadata const& rhs) {
   os << "NotificationMetadata={id=" << rhs.id();
 
   // custom_attributes()
-  char const* sep = "custom_attributes.";
-  for (auto const& kv : rhs.custom_attributes()) {
-    os << sep << kv.first << "=" << kv.second;
-    sep = ", custom_attributes.";
-  }
+  os << "custom_attributes.";
```
Comment (devbww) [inline google/cloud/storage/notification_metadata.cc:59]: This was conditional before.

File: google/cloud/bigtable/examples/bigtable_examples_common.cc (hunk 1 above)
Comment (remyabel) [reply google/cloud/bigtable/examples/bigtable_examples_common.cc]: Good catch, thanks.

File: google/cloud/storage/notification_metadata.cc (hunk 2 above)
Comment (remyabel) [reply google/cloud/storage/notification_metadata.cc]: Fixed I think.

File: google/cloud/bigtable/examples/bigtable_examples_common.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/bigtable/examples/bigtable_examples_common.cc]: It looks like they are now different when `args` is empty.

File: google/cloud/bigtable/examples/bigtable_examples_common.cc [hunk 3]
```diff
@@ -171,10 +172,8 @@ google::cloud::bigtable::examples::Commands::value_type MakeCommandEntry(
         argv.size() != 3 + args.size()) {
       std::ostringstream os;
       os << name << " <project-id> <instance-id> <table-id>";
-      char const* sep = " ";
-      for (auto const& a : args) {
-        os << sep << a;
-      }
+      if (!args.empty()) os << " ";
+      os << absl::StrJoin(args, " ");
```
Comment (devbww) [inline google/cloud/bigtable/examples/bigtable_examples_common.cc:176]: Looks good to me.

Nit: For my money, I'd coalesce the ones that now test `empty()`.  For example,
```
  if (!args.empty()) os << ' ' << absl::StrJoin(args, ' ');
```

Comment (remyabel) [reply google/cloud/bigtable/examples/bigtable_examples_common.cc]: Agreed.

File: google/cloud/storage/notification_metadata.cc [hunk 4]
```diff
@@ -55,21 +56,15 @@ std::ostream& operator<<(std::ostream& os, NotificationMetadata const& rhs) {
...
 
   // custom_attributes()
-  char const* sep = "custom_attributes.";
-  for (auto const& kv : rhs.custom_attributes()) {
-    os << sep << kv.first << "=" << kv.second;
-    sep = ", custom_attributes.";
-  }
+  if (!rhs.custom_attributes().empty()) os << "custom_attributes.";
+  os << absl::StrJoin(rhs.custom_attributes(), ", custom_attributes.",
+                      absl::PairFormatter("="));
```
Comment (devbww) [inline google/cloud/storage/notification_metadata.cc:61]: This one too, perhaps?

**Review by devbww — APPROVED** [review]

**Review by coryan — APPROVED** [review]
:lgtm: let's see if the builds agree +kokoro:run 

> Reviewed 6 of 8 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5159#-:-MI4z2Ru50H4bVraxJe9:bgp7kb5)* status: all files reviewed, 4 unresolved discussions (waiting on @devbww and @remyabel)

**Review by coryan — CHANGES_REQUESTED** [review]
@remyabel there is one additional comment below, do you mind fixing it too?

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5159#-:-MI533DD6vjCKEKJLEg6:bn1x1i)* status: all files reviewed, 4 unresolved discussions (waiting on @devbww and @remyabel)

File: google/cloud/storage/notification_metadata.cc (hunk 4 above)
Comment (remyabel) [reply google/cloud/storage/notification_metadata.cc]: Fixed.

**Review by coryan — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5159#-:-MI5GHLE67U87_LfqZFh:bnfp4nl)* status: all files reviewed, 4 unresolved discussions (waiting on @devbww and @remyabel)

**Pull request merged**